frontend/src/features/admin-form/common/AdminViewFormService.ts

```diff
@@ -47,7 +54,19 @@ export const previewForm = async (
 ): Promise<PreviewFormViewDto> => {
   return ApiService.get<PreviewFormViewDto>(
     `${ADMIN_FORM_ENDPOINT}/${formId}/preview`,
-  ).then(({ data }) => data)
+  ).then(({ data }) => {
+    const mockedSpcpSession: PreviewFormViewDto['spcpSession'] =
+      data.spcpSession ?? data.form.authType !== FormAuthType.NIL
+        ? { userName: PREVIEW_MOCK_UINFIN }
+        : undefined
```

Comment:
This code is not assigning `data.spcpSession` to mockedSpcpSession when `data.spcpSession` is set, is that intentional?

I had some difficulty reading this statement btw, with contending with the 3 operators precedence (`??`, `!==`, and `?:`). Perhaps we could split this into multiple statements, or add braces to emphasize intent 🤔 

Like, I wasn't sure if the intent of the code was to write this behaviour:
```javascript
const spcpSessionIsSet = data.spcpSession !== undefined && data.spcpSession !== null

mockedSpcpSession = undefined

if (spcpSessionIsSet) {
    mockedSpcpSession = data.spcpSession
}
else if (data.form.authType !== FormAuthType.NIL) {
    mockedSpcpSession = { userName: PREVIEW_MOCK_UINFIN }
}
```

or this: (which is the current behaviour)

```javascript
const spcpSessionIsSet = data.spcpSession !== undefined && data.spcpSession !== null
const spcpSessionIsTruthy = !!data.spcpSession

mockedSpcpSession = undefined

if (spcpSessionIsSet && spcpSessionIsTruthy) {
    mockedSpcpSession = { userName: PREVIEW_MOCK_UINFIN }
}
else if (data.form.authType !== FormAuthType.NIL) {
    mockedSpcpSession = { userName: PREVIEW_MOCK_UINFIN }
}
```

Just as an additional side note, I was confused by the `??` because it make it seems like we have a concern for a case of defined but falsy, but since `data.spcpSession` is of type SpcpSession (an object), it's not a valid concern 🤔

Reply:
yea, the data sent by the server should take precedence (but as of now nothing will be sent by the server since there is probably no need to).

will update to more it more readable

Reply:
fixed in d8b3feb